packages/jwt-service/src/types/IJwtService.ts

```diff
@@ -164,4 +164,4 @@ export function isJwsJsonGeneral(jws: Jws): jws is JwsJsonGeneral {
   return typeof jws === 'object' && 'signatures' in jws
 }
 
-export const COMPACT_JWS_REGEX = /^([a-zA-Z0-9_=-]+)\.([a-zA-Z0-9_=-]+)?\.([a-zA-Z0-9_=-]+)$/
+export const COMPACT_JWS_REGEX = /^([a-zA-Z0-9_=-]+)\.([a-zA-Z0-9_=-]+)?(\.([a-zA-Z0-9_=-]+)?)?\.?$/
```

Comment:
Why not change the last + to * ?

Reply:
I did not permit more than the extra dot.  Wouldn't changing the last + to * would capture the third part as empty string possibly increasing parts.length to 3?
I did not test that so I just added an optional dot.

Reply:
Even worse, I believe this regex playnly is wrong now as it will probably capture the wrong things in group 3 because you added two time a ? in there as well

Reply:
Just tested it quickly in an online regex tester, and indeed it is way of now

Reply:
Group 3 including a . all of a sudden. It starts accepting strings with 4 dots in there etc.

Reply:
I updated to be * and tested in regex101

Reply:
```^([a-zA-Z0-9_=-]+)\.([a-zA-Z0-9_=-]+)?\.([a-zA-Z0-9_=-]+)?$```
If there is no payload so ey..ey then there is no group 2.  IF there is no signature so ey.ey. then there is no group 3. Seems to accomplish what you are after

Reply:
See my remark, because my initial suggestion returns an empty group 3, whilst the latest is in line with the initial expectation

Reply:
This does not allow extra groups like group 4
^([a-zA-Z0-9_=-]+)\.([a-zA-Z0-9_=-]+)(?:\.([a-zA-Z0-9_=-]+))?\.?$

However it does allow a 4th dot, but as soon as I type an extra char it stops matching
eyJ0eXAiOiJrYitqd3QiLCJhbGciOiJFUzI1NiJ9.eyJpYXQiOjE3MjcwODM2ODA4MzMsIm5vbmNlIjoiYjkzNzM2NzgtOTlhNi00NzYxLTk0Y2ItZWJmODMzZWQ0NTY4Iiwic2RfaGFzaCI6Il9icG5UMkdzNnZ4VTlxTWJhNkZDbnM5RGNVd2U5dXhldmlRUjNaY1d0a1EifQ.jlvRRDBXOmlVhDuhhJQ8GWF_LIAUtDEbUNWTNAtdpvBTNM2HsTT4QuM6zG7F2Jj1D0ri8qOzC1sGoslgNwcuYw.

Reply:
Yes which is invalid. Again see the regex in my last remark

Reply:
Your last remark says
^([a-zA-Z0-9_=-]+)\.([a-zA-Z0-9_=-]+)?\.([a-zA-Z0-9_=-]+)?$
That works, but for an unsigned one the second dot is now mandatory
eyJ0eXAiOiJrYitqd3QiLCJhbGciOiJFUzI1NiJ9.eyJpYXQiOjE3MjcwODM2ODA4MzMsIm5vbmNlIjoiYjkzNzM2NzgtOTlhNi00NzYxLTk0Y2ItZWJmODMzZWQ0NTY4Iiwic2RfaGFzaCI6Il9icG5UMkdzNnZ4VTlxTWJhNkZDbnM5RGNVd2U5dXhldmlRUjNaY1d0a1EifQ.
when I take the last dot it fails. Actually that's how it's in the spec, but we do loose some liency.

Reply:
tbc ^([a-zA-Z0-9_=-]+).([a-zA-Z0-9_=-]+)?.([a-zA-Z0-9_=-]+)?$ is now what's in the last commit